src/worker/utils/chem.ts

```diff
@@ -70,9 +70,21 @@ export const getMoleculeDetails = (smiles: string) => {
   };
 };
 
-export const getCanonicalFormForStructure = (structure: string): string | null => {
-  if (isValidSmiles(structure)) return getCanonicalSmiles(structure);
-  return getCanonicalSmarts(structure);
+export const getCanonicalFormForStructure = ({
+  structure,
+  molNotation = 'smiles',
+  useQMol = false,
+}: {
+  structure: string;
+  molNotation?: MolNotation;
+  useQMol?: boolean;
```

Comment:
Do we need both? can we infer `useQMol` by doing `const useQMol = molNotation === 'smarts'` ?

Reply:
If im not wrong, you can also create `qmol` from a smiles or molblock, and also create `mol` from a smarts (if it works), so better let the user choose.